rust/src/encodings/plain.rs

```diff
@@ -99,9 +98,13 @@ impl<'a> PlainEncoder<'a> {
             self.encode_boolean(boolean_arr).await?;
         } else {
             let byte_width = array.data_type().byte_width();
-            self.writer
-                .write_all(&data.buffers()[0].as_slice()[0..array.len() * byte_width])
-                .await?;
+            let slice = unsafe {
+                from_raw_parts(
+                    data.buffers()[0].as_ptr().add(data.offset() * byte_width),
+                    data.len() * byte_width,
```

Comment:
what's the diff between array.len() vs data.len() ?

Reply:
Oh no real difference, was looking at 

https://docs.rs/arrow-array/33.0.0/src/arrow_array/array/primitive_array.rs.html#308-311